src/test/test_flake8.py

```diff
@@ -19,8 +19,13 @@
 @pytest.mark.flake8
 @pytest.mark.linter
 def test_flake8():
-    ignore_files = 'src/caret_analyze/architecture/struct/__init__.py'
-    rc, errors = main_with_errors(argv=[f'--exclude={ignore_files}'])
-    assert rc == 0, 'Found %d code style errors / warnings:\n' % len(
-        errors
-    ) + '\n'.join(errors)
+    rc, errors = main_with_errors()
+
+    filtered_errors = []
+    for error in errors:
+        if 'A005' not in error and 'I300' not in error:
+            filtered_errors.append(error)
```

Comment:
Currently, the error codes 'A005' and 'I300' are hardcoded in the filtering logic. To make the code more maintainable and scalable (in case more rules need to be ignored later), consider refactoring as follows:

```suggestion
    ignored_rules = ['A005', 'I300']

    filtered_errors = []
    for error in errors:
        if not any(rule in error for rule in ignored_rules):
            filtered_errors.append(error)
```